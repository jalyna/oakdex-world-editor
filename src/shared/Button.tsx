import * as React from 'react'
import styled from 'styled-components'

import { DEFAULT_FONT, GREY_50, GREY_90, GREY_70, TEAL_30 } from 'shared/theme'

interface ButtonProps {
  isActive?: boolean
}

const StyledButton = styled.button`
  cursor: pointer;
  font-family: ${DEFAULT_FONT};
  font-size: 16px;
  color: ${({ isActive }: ButtonProps) => isActive ? 'white' : GREY_50};
  border: 1px solid ${({ isActive }: ButtonProps) => isActive ? TEAL_30 : GREY_90};
  padding: 5px 10px;
  outline: none;
  background: ${({ isActive }: ButtonProps) => isActive ? TEAL_30 : 'white'};
  border-radius: 4px;
  box-sizing: border-box;
  width: 100%;

  &:hover {
    border-color: ${GREY_70};
  }

  &:disabled {
    opacity: 0.6;
    cursor: default;
  }
`

export default function Button (props: any) {
  const isActive = props.isActive as boolean
  return <StyledButton {...props} />
}
