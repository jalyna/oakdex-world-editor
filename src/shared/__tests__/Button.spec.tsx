import * as React from 'react'
import { mount } from 'enzyme'
import Button from '../Button'

describe('<Button>', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Button onClick={jest.fn()} />)
    expect(wrapper).toMatchSnapshot()
  })
})
