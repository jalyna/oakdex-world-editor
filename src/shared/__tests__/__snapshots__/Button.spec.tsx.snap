// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders correctly 1`] = `
<Button
  onClick={[MockFunction]}
>
  <styled.button
    onClick={[MockFunction]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "dJicAm",
            "rules": Array [
              "
  cursor: pointer;
  font-family: ",
              "'Roboto', sans-serif",
              ";
  font-size: 16px;
  color: ",
              [Function],
              ";
  border: 1px solid ",
              [Function],
              ";
  padding: 5px 10px;
  outline: none;
  background: ",
              [Function],
              ";
  border-radius: 4px;
  box-sizing: border-box;
  width: 100%;
  ",
              [Function],
              ";

  &:hover {
    border-color: ",
              "#b3b3b3",
              ";
  }

  &:disabled {
    opacity: 0.6;
    cursor: default;
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[MockFunction]}
    >
      <button
        className="sc-bdVaJa dJicAm"
        onClick={[MockFunction]}
      />
    </StyledComponent>
  </styled.button>
</Button>
`;
